tree/treeplayer/test/dataframe/vecops_tvec.cxx

```diff
@@ -0,0 +1,238 @@
+   }
+
+   // The same with 1 view
+   ROOT::Experimental::VecOps::TVec<double> w(ref.data(), ref.size());
+   plus = w + vec;
+   minus = w - vec;
+   mult = w * vec;
+   div = w / vec;
+
+   for (unsigned int i = 0; i < v.size(); ++i) {
+      EXPECT_EQ(plus[i], ref[i] + vec[i]);
+      EXPECT_EQ(minus[i], ref[i] - vec[i]);
+      EXPECT_EQ(mult[i], ref[i] * vec[i]);
+      EXPECT_EQ(div[i], ref[i] / vec[i]);
+   }
+
+   // The same with 2 views
+   ROOT::Experimental::VecOps::TVec<double> w2(ref.data(), ref.size());
+   plus = w + w2;
+   minus = w - w2;
+   mult = w * w2;
+   div = w / w2;
+
+   for (unsigned int i = 0; i < v.size(); ++i) {
+      EXPECT_EQ(plus[i], ref[i] + w2[i]);
+      EXPECT_EQ(minus[i], ref[i] - w2[i]);
+      EXPECT_EQ(mult[i], ref[i] * w2[i]);
+      EXPECT_EQ(div[i], ref[i] / w2[i]);
+   }
+}
+
+TEST(VecOps, MathVectorInPlace)
+{
+   using namespace ROOT::Experimental::VecOps;
+   TVec<double> ref{1, 2, 3};
+   TVec<double> vec{3, 4, 5};
+   TVec<double> v(ref);
+   auto plus = v;
+   plus += vec;
+   auto minus = v;
+   minus -= vec;
+   auto mult = v;
+   mult *= vec;
+   auto div = v;
+   div /= vec;
+
+   for (unsigned int i = 0; i < v.size(); ++i) {
+      EXPECT_EQ(plus[i], ref[i] + vec[i]);
+      EXPECT_EQ(minus[i], ref[i] - vec[i]);
+      EXPECT_EQ(mult[i], ref[i] * vec[i]);
+      EXPECT_EQ(div[i], ref[i] / vec[i]);
+   }
+}
+
+TEST(VecOps, Filter)
+{
+   using namespace ROOT::Experimental::VecOps;
+   TVec<int> v{0, 1, 2, 3, 4, 5};
+   std::vector<int> vEvenRef{0, 2, 4};
+   std::vector<int> vOddRef{1, 3, 5};
+   auto vEven = v[v % 2 == 0];
+   auto vOdd = v[v % 2 == 1];
+   for (int i = 0; i < 3; ++i) {
+      EXPECT_EQ(vEven[i], vEvenRef[i]);
+      EXPECT_EQ(vOdd[i], vOddRef[i]);
+   }
+}
+
+template <typename T, typename V>
+void CompAndPrintTVec(ROOT::Experimental::VecOps::TVec<T> v, V w)
+{
+   using namespace ROOT::Experimental::VecOps;
+   std::cout << v << " " << w << std::endl;
+   std::cout << v + w << std::endl;
+   std::cout << v - w << std::endl;
+   std::cout << v * w << std::endl;
+   std::cout << v / w << std::endl;
+   std::cout << (v > w) << std::endl;
+   std::cout << (v >= w) << std::endl;
+   std::cout << (v == w) << std::endl;
+   std::cout << (v <= w) << std::endl;
+   std::cout << (v < w) << std::endl;
+}
+
+TEST(VecOps, PrintOps)
+{
+   using namespace ROOT::Experimental::VecOps;
+   TVec<int> ref{1, 2, 3};
+   TVec<int> v(ref);
+   CompAndPrintTVec(v, 2.);
+   CompAndPrintTVec(v, ref + 2);
+
+   ROOT::Experimental::VecOps::TVec<int> w(ref.data(), ref.size());
+
+   CompAndPrintTVec(v, 2.);
+   CompAndPrintTVec(v, ref + 2);
+}
+
+void CheckEq(std::string_view name, const ROOT::Experimental::VecOps::TVec<double> v,
+             const ROOT::Experimental::VecOps::TVec<double> w)
+{
+   auto wsize = w.size();
+   EXPECT_EQ(wsize, v.size());
+   for (unsigned int i = 0; i < wsize; i++) {
+      EXPECT_DOUBLE_EQ(v[i], w[i]) << " error while checking math function " << name;
+   }
```

Comment:
seems like we are missing `All` and `Any`: this should be
```c++
EXPECT_TRUE(All(v[i] == w[i]));
```
or to be super correct `AllClose(v[i] == w[i])`.

Reply:
Sorry I do not get this: what do you mean?

Reply:
the fact that you had to write 4 lines to check that all elements of one TVec were equal to all elements of another shows that we are missing some useful helpers: `All`, `Any` and `AllClose` (stealing the numpy denominations)

Reply:
Oh, now I see. Fair enough.

Reply:
std::equal is probably to be used, or a wrapper around it.